liberapay/i18n/base.py

```diff
@@ -272,6 +272,10 @@ def format_timedelta(self, o, **kw):
     def parse_money_amount(self, string, currency, maximum=D_MAX):
         group_symbol = self.number_symbols['group']
         decimal_symbol = self.number_symbols['decimal']
+        # If first character is relavent currency symbol, remove it and pass only numeric string
+        # InvalidNumber will catch error if using the wrong currency symbol
+        if string.startswith(self.currency_symbols.get(currency, currency)):
+            string = string[1:]
```

Comment:
I didn't notice before, but this is incorrect. Firstly because the currency symbol isn't always a single character. Secondly because the previous line falls back to the three-letter ISO currency code when there is no symbol.

I also see two needed improvements:

1. We should also look for the currency symbol at the end of the string, because in most languages it's standard to put the currency symbol after the number, not before.
2. Strip the string, both before looking for the currency symbol and after it has been removed, because people put spaces everywhere, sometimes without realizing it.

Reply:
Ok. I'll start working on that. By the way, how do I see what going on in the Python backend? For instance if I wanted the print the value of `string` at some point? I'm not experienced in Python on the web. I'm also pretty new to makefiles. Does `make run` start some kind of local Python server? Or is it something else?

Reply:
When you do a `make run`, two commands are executed, which you can see in the output:

1. The first one applies database migrations: `PYTHONPATH=. env/bin/python cli/run.py -e defaults.env,local.env env/bin/python liberapay/models/__init__.py`
2. The second one runs `app.py` (which launches the [Gunicorn](https://gunicorn.org/) HTTP server): `PATH=env/bin:$PATH env/bin/python cli/run.py -e defaults.env,local.env env/bin/python app.py`

Both commands first call `cli/run.py` to load environment variables from the `defaults.env` and `local.env` files.

If you add a `print(string)` line in the code, then the value will show up in your terminal. If the code is run through Gunicorn, then the printed value will be followed by at least one HTTP request log line, so don't expect it to be the last line of the output. If the code is run through pytest, then the printed value will be in the `Captured stdout call` section of the output for each failing test that encounters your print call.

Reply:
This is what I've come up with. I've tested it with currency symbols of different lengths, with the symbol in the front and back, and with spaces between the numerical amount and the symbol, and at either end of the whole string.

```python
# If string begins or ends with relavent currency symbol, remove it and pass only numeric string
# InvalidNumber will catch error if using the wrong currency symbol
        string = string.strip()
        currency_length = len(self.currency_symbols.get(currency, currency))
        if string.startswith(self.currency_symbols.get(currency, currency)):
            string = string[currency_length:]
        elif string.endswith(self.currency_symbols.get(currency, currency)):
            string = string[0:-currency_length]
        string = string.strip()
```